core/src/main/java/org/apache/brooklyn/core/mgmt/persist/BrooklynMementoPersisterToObjectStore.java

```diff
@@ -294,7 +294,7 @@ public void visit(BrooklynObjectType type, String id, String contentsSubpath) th
                     exceptionHandler.onLoadMementoFailed(type, "memento "+id+" read error", e);
                 }
                 
-                String xmlId = (String) XmlUtil.xpath(contents, "/"+type.toCamelCase()+"/id");
+                String xmlId = (String) XmlUtil.xpathHandlingIllegalChars(contents, "/"+type.toCamelCase()+"/id");
```

Comment:
We're making the assumption that the `id` field etc won't have crazy characters in it (because the Brooklyn framework generates those ids). The place we can get crazy characters is in attributes and config values that are set by the user.